patches/common/chromium/frame_host_manager.patch

```diff
@@ -107,8 +141,36 @@ index 2314c6d1d29ebc61d5156644617e9eb088df5d43..9c75367a173b836027b3b7628b49b363
    return dest_site_instance;
  }
  
+diff --git a/content/browser/site_instance_impl.cc b/content/browser/site_instance_impl.cc
+index 703409b11a43f94896dc8b74eb87e3bfc7313710..ed5e84980f12cb86634b24a6d73d343a3308ee2a 100644
+--- a/content/browser/site_instance_impl.cc
++++ b/content/browser/site_instance_impl.cc
+@@ -297,6 +297,11 @@ scoped_refptr<SiteInstance> SiteInstanceImpl::GetRelatedSiteInstance(
+       url, /* allow_default_instance */ true);
+ }
+ 
++scoped_refptr<SiteInstance> SiteInstanceImpl::CreateRelatedSiteInstance(const GURL& url) {
++  return browsing_instance_->CreateSiteInstanceForURL(url);
```

Comment:
Why can't this use `GetSiteInstanceForURL` in the existing browsing instance ? `browsing_instance_->GetSiteInstanceForURL(url, false /* allow_default_instance */);`

with the current implementation in the PR, any call to create the site instance for same url will go untracked by the site instance map maintained by browsing instance. I don't see an issue with reusing the site instance from the browsing instance, since we are maintaining a 1:1 relation.

Reply:
> Why can't this use `GetSiteInstanceForURL` in the existing browsing instance ? `browsing_instance_->GetSiteInstanceForURL(url, false /* allow_default_instance */);`

`browsing_instance_->GetSiteInstanceForURL` [is equivalent](https://cs.chromium.org/chromium/src/content/browser/site_instance_impl.cc?type=cs&sq=package:chromium&g=0&l=336-340) to the [first suggestion](https://github.com/electron/electron/pull/18173/commits/50aae0b0bf8612dfdec8c2c75493dc09818d3e3f) of this PR.
As [discussed earlier in the thread](https://github.com/electron/electron/pull/18173#pullrequestreview-235362945), it can't work because, for the same URL (during a reload for instance), we'll end up reusing the same `SiteInstance`, and, as a result, the render process won't be restarted — in opposition to what is expected.

> with the current implementation in the PR, any call to create the site instance for the same url will go untracked by the site instance map maintained by browsing instance

I'm not sure because in `CreateSiteInstanceForURL`, I call `instance->SetSite` that calls `RegisterSiteInstance` which adds the `SiteInstance` to the ` site_instance_map_` of browsing instance.

>  I don't see an issue with reusing the site instance from the browsing instance since we are maintaining a 1:1 relation.

As I understood with @zcbenz , reusing the site instance will not trigger render process restart which is an issue. Maybe I have not understood this last sentence @deepak1556 though

Reply:
Yes we need to restart process. Regarding the site instance map statement, when `GetSiteInstanceForURL` it will either create a new instance for the url, add it to the map and return it, otherwise return an existing one from the map. With this PR implementation, a site instance is created and add to the map but when a new request is made for the same url, it will create the instance but will not add it to the map, check the condition in `BrowsingInstance::RegisterSiteInstance`. I was concerned about this.

But since we are using the same browser instance makes sense that `GetSiteInstanceForURL` can't be used to force restart the process.

Reply:
> Even if we did not register it, I don't see any issue with that so far, since we want to use new SiteInstance every time.

Browsing Instance has a reference to the site instance when its created, if its not registered we will be managing the lifetime.